Sources/RemoteMessaging/RemoteMessagingStore.swift

```diff
@@ -0,0 +1,383 @@
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Common
+import CoreData
+import BrowserServicesKit
+import Persistence
+
+public enum RemoteMessagingStoreError: Error {
+    case saveConfigFailed
+    case invalidateConfigFailed
+    case updateMessageShownFailed
+    case saveMessageFailed
+    case updateMessageStatusFailed
+    case deleteScheduledMessageFailed
+}
+
+public final class RemoteMessagingStore: RemoteMessagingStoring {
+
+    public struct Notifications {
+        static public let remoteMessagesDidChange = Notification.Name("com.duckduckgo.app.RemoteMessagesDidChange")
+    }
+
+    private enum RemoteMessageStatus: Int16 {
+        case scheduled
+        case dismissed
+        case done
+    }
+
+    public enum Constants {
+        public static let privateContextName = "RemoteMessaging"
+    }
+
+    let context: NSManagedObjectContext
+    let notificationCenter: NotificationCenter
+
+    public convenience init(
+        database: CoreDataDatabase,
+        notificationCenter: NotificationCenter = .default,
+        errorEvents: EventMapping<RemoteMessagingStoreError>?,
+        log: @escaping @autoclosure () -> OSLog = .disabled
+    ) {
+        self.init(
+            context: database.makeContext(concurrencyType: .privateQueueConcurrencyType, name: Constants.privateContextName),
+            notificationCenter: notificationCenter,
+            errorEvents: errorEvents,
+            log: log()
+        )
+    }
+
+    public init(
+        context: NSManagedObjectContext,
+        notificationCenter: NotificationCenter = .default,
+        errorEvents: EventMapping<RemoteMessagingStoreError>?,
+        log: @escaping @autoclosure () -> OSLog = .disabled
+    ) {
+        self.context = context
+        self.notificationCenter = notificationCenter
+        self.errorEvents = errorEvents
+        self.getLog = log
+    }
+
+    public func saveProcessedResult(_ processorResult: RemoteMessagingConfigProcessor.ProcessorResult) {
+        os_log("Remote messaging config - save processed version: %d", log: log, type: .debug, processorResult.version)
+        saveRemoteMessagingConfig(withVersion: processorResult.version)
+
+        if let remoteMessage = processorResult.message {
+            deleteScheduledRemoteMessages()
+            save(remoteMessage: remoteMessage)
+
+        } else {
+            deleteScheduledRemoteMessages()
+        }
+
+        DispatchQueue.main.async {
+            self.notificationCenter.post(name: Notifications.remoteMessagesDidChange, object: nil)
+        }
+    }
+
+    private let errorEvents: EventMapping<RemoteMessagingStoreError>?
+
+    private let getLog: () -> OSLog
+    private var log: OSLog {
+        getLog()
+    }
+}
+
+// MARK: - RemoteMessagingConfigManagedObject Public Interface
+
+extension RemoteMessagingStore {
```

Comment:
All the code below this line has been directly copied over from iOS, with minor adjustments, such as using convenience `fetchRequest()` methods and initializers of NSManagedObject subclasses, as well as using RemoteMessagingConfig convenience initializer with the managed object.